src/librustc_front/hir.rs

```diff
@@ -1226,7 +1248,7 @@ pub enum Item_ {
     /// An enum definition, e.g. `enum Foo<A, B> {C<A>, D<B>}`
     ItemEnum(EnumDef, Generics),
     /// A struct definition, e.g. `struct Foo<A> {x: A}`
-    ItemStruct(P<StructDef>, Generics),
+    ItemStruct(P<VariantData>, Generics),
```

Comment:
@petrochenkov 

The policy is that you try not to long-pole things. `Item` is 160 bytes and `Generics` is 72 bytes so we have plenty of space for `VariantData`.